pkg/asset/machines/azure/machines.go

```diff
@@ -280,6 +295,28 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 	return spec, nil
 }
 
+func getBootDiagnosticObject(diag *azure.BootDiagnostics, cloudName azure.CloudEnvironment, role string) (bootDiagnostics *machineapi.AzureDiagnostics) {
```

Comment:
```suggestion
func getBootDiagnosticObject(diag *azure.BootDiagnostics, cloudName azure.CloudEnvironment, role string) (*machineapi.AzureDiagnostics) {
```
I don't think we need a named return value here